helm/resource_release.go

```diff
@@ -541,9 +541,10 @@ func resourceReleaseCreate(ctx context.Context, d *schema.ResourceData, meta int
 			return diag.FromErr(err)
 		}
 
-		debug("%s Release was created but returned an error", logID)
+		debug("%s Release was created but returned an error, rolling back", logID)
```

Comment:
I would propose adding a warning diagnostic here to let the user know that this has happened. 

We perhaps shouldn't use the "rolling back" terminology as we're not actually doing the same thing as a `helm rollback`. Perhaps saying that the release was created in a failed state and was therefore deleted.